lib/order_checker.rb

```diff
@@ -0,0 +1,46 @@
+require_relative 'send_text'
+
+class OrderChecker
+
+  def initialize(menu, order_array, send_text = SendText.new)
+    @menu = menu
+    @order_array = order_array
+    @send_text = send_text
+  end
+
+  def check_order
+    check_if_valid
+    check_if_correct_total
+    @send_text.send
+  end
+
+  private
+
+  def check_if_valid
+    raise "Invalid dish selection!" unless valid_order?
+  end
+
+  def valid_order?
+    @order_array[0].each { |i| return false if i[0] >= (@menu.length + 1) }
+    true
+  end
+
+  def check_if_correct_total
+    given_total = @order_array[1]
+    compare_totals(given_total, calculate_total)
+  end
+
+  def calculate_total
+    total = 0
+    @order_array[0].each do |order|
+      total += @menu[order[0] - 1][:price] * order[1]
```

Comment:
Also, [order[0] - 1] is confusing. I think there are two problems here:

1) You've offset the index of the order item from it's index in the array by one. If this has been done to make it easier for the user when placing the order, fine, but you should deal with the offset at the entry point for order. So if I ask for 10 of item 1, then in the place_order method I should translate that to 10 of item index 0 before passing the data on to other classes - user interface and business logic are two separate responsibilities and your business logic shouldn't be more opaque because of decisions that were made for user interaction. 

2) It's not clear from reading this code what order[0] even is in the first place. When making the switch above, why not just go the whole distance and substitute the input code for the corresponding items? So something like:

```
def place_order(order_input)
  order = build_order(order_input)
  checker = @order_checker.new(@menu, order)
  .. etc
end

def build_order(order_input)
  @order_array[0].map do {
    menu_index = order_line[0]
    item_quantity = order_line[1]
    [ :item => menu[menu_index], :quantity => item_quantity ]
  }.to_h
end
```